### DISCUSSION TITLE
Routing based on header

### DISCUSSION THREAD

**Discussion by kaustubhsharma-009** [Kong/kong#7781] [Help] [answered]
Hi all, 
I'm new to this service. Just wanted to know how would we route traffic dynamically based on a header in the request -> something like stick tables in HAProxy or sticky sessions in nginx

**Comment by dndx** [comment, marked as answer]
Hello @kaustubhsharma-009,

You should probably give a look at the `headers` field of the `Route` object of Kong for simple route by header use cases:

https://docs.konghq.com/gateway-oss/2.5.x/admin-api/#add-route

For more complex use cases, you may want to refer to the [Route by header](https://docs.konghq.com/hub/kong-inc/route-by-header/) plugin that is available as part of our enterprise offering.
